segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -450,18 +450,27 @@ public boolean isOffline() {
         SegmentMetadata sourceMetadata = this.metadata.getStreamSegmentMetadata(sourceSegmentId);
 
         CompletableFuture<Void> result;
-        if (sourceMetadata.isSealed()) {
-            // Source Segment is already sealed, so we only need to merge it.
-            result = this.durableLog.add(new MergeSegmentOperation(targetSegmentId, sourceSegmentId), timer.getRemaining());
+        if (sourceMetadata.getLength() == 0) {
```

Comment:
Yes, it shows up in the if branch as well, but there are some differences:
1. The point of the outer one is to see if there is some data already in the transaction. If so, then pipeline the seal and merge operations together (as opposed from running them in sequence - pipelining is faster).
2. The purpose of the inner one (in the if block), is to verify if there was any outstanding data in the pipeline while we were checking the first time. After the seal operation is done, no further changes are allowed to the segment, so whatever value we have there is the final value. 

I could completely eliminate the `else` block and just serialize the two operations, but I want to optimize for the most common scenarios, which is when we actually do have some data in the transaction.

I did do a small improvement though; I moved the `trySealStreamSegment` out before the if block, since it was the first statement in both the if and the else blocks.